roles/kubespray-defaults/defaults/main/download.yml

```
@@ -995,6 +1000,18 @@ downloads:
     groups:
       - kube_node
 
+  gateway_api_crds:
+    enabled: "{{ gateway_api_enabled }}"
+    file: true
+    version: "{{ gateway_api_version }}"
+    dest: "{{ local_release_dir }}/gateway-api-{{ gateway_api_channel }}-install.yaml"
+    checksum: "{{ gateway_api_crds_checksums.no_arch[gateway_api_channel][gateway_api_version] }}"
```

If going with the "top level checksums" here you could use :
```suggestion
    checksum: "{{ lookup('vars', 'gateway_api_' + gateway_api_channel + '_crds_checksums').no_arch[gateway_api_version] }}"
```

---

That's cool!

---

Yeah lookup can do wonderful things : https://github.com/ansible-collections/community.general/issues/9107#issuecomment-2468338199